sway/tree/container.c

```diff
@@ -1609,3 +1619,68 @@ bool container_is_scratchpad_hidden_or_child(struct sway_container *con) {
 	con = container_toplevel_ancestor(con);
 	return con->scratchpad && !con->workspace;
 }
+
+static enum sway_container_layout layout_category(enum sway_container_layout layout) {
+	switch (layout) {
+		case L_HORIZ: /* fallthrough */
+		case L_TABBED:
+			return L_HORIZ;
+		case L_VERT: /* fallthrough */
+		case L_STACKED:
+			return L_VERT;
+		case L_NONE:
+			return L_NONE;
+	}
+	return L_NONE;
+}
+
+static bool container_is_squashable(struct sway_container *con,
+		struct sway_container *child) {
+	enum sway_container_layout gp_orient = layout_category(
+			con->parent ? con->parent->layout : con->workspace->layout);
+	enum sway_container_layout con_orient = layout_category(con->layout);
+	enum sway_container_layout child_orient = layout_category(child->layout);
+
+	return (con->layout == L_HORIZ || con->layout == L_VERT) &&
+		(child->layout == L_HORIZ || child->layout == L_VERT) &&
+		con_orient != child_orient && gp_orient == child_orient;
+}
+
+static void container_squash_children(struct sway_container *con) {
+	for (int i = 0; i < con->children->length; i++) {
+		struct sway_container *child = con->children->items[i];
+		i += container_squash(child);
+	}
+}
+
+// Returns the number of new containers added to the parent
+int container_squash(struct sway_container *con) {
+	int change = 0;
+	if (!con->children) {
+		return change;
+	}
+	if (con->children->length != 1) {
+		container_squash_children(con);
+		return change;
```

Comment:
Looks like this is also always 0; let's make that explicit here and move the `int change = 0;` declaration to L1668.

Reply:
Seems good. Technically I could just declare it as `int change = -1` instead of ending with `change--` since I know in advance I will reap con. I'll just keep it as 0 though I think. I prefer that the value matches its meaning throughout.